lib/models/authentication/authentication_service.dart

```diff
@@ -2,20 +2,48 @@ import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/services.dart';
 import 'package:google_sign_in/google_sign_in.dart';
 import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
-import '../../data/data.dart';
+import 'package:hatspace/data/data.dart';
+import 'package:hatspace/models/storage/storage_service.dart';
 import 'authentication_exception.dart';
 
+enum SignUpType { googleService, facebookService }
+
 class AuthenticationService {
   final GoogleSignIn _googleSignIn;
   final FirebaseAuth _firebaseAuth;
   final FacebookAuth _facebookAuth;
+  final StorageService _storageService;
   AuthenticationService({
     GoogleSignIn? googleSignIn,
     FirebaseAuth? firebaseAuth,
     FacebookAuth? facebookAuth,
+    StorageService? storageService,
   })  : _googleSignIn = googleSignIn ?? GoogleSignIn(),
         _firebaseAuth = firebaseAuth ?? FirebaseAuth.instance,
-        _facebookAuth = facebookAuth ?? FacebookAuth.instance;
+        _facebookAuth = facebookAuth ?? FacebookAuth.instance,
+        _storageService = storageService ?? StorageService();
+
+  Future<List<Roles>> signUp({SignUpType? signUpType}) async {
```

Comment:
this is business logic, please keep it in bloc. Within authentication service, it does not need to know user's role